homeassistant/components/alexa/smart_home.py

```diff
@@ -383,12 +377,84 @@ def get_property(self, name):
             raise _UnsupportedProperty(name)
 
         unit = self.entity.attributes[CONF_UNIT_OF_MEASUREMENT]
+        temp = self.entity.state
+        if self.entity.domain == climate.DOMAIN:
+            temp = self.entity.attributes.get(
+                climate.ATTR_CURRENT_TEMPERATURE, 0)
         return {
-            'value': float(self.entity.state),
+            'value': float(temp),
             'scale': API_TEMP_UNITS[unit],
         }
 
 
+class _AlexaThermostatController(_AlexaInterface):
+    def name(self):
+        return 'Alexa.ThermostatController'
+
+    def properties_supported(self):
+        properties = []
+        supported = self.entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE:
+            properties.append({'name': 'targetSetpoint'})
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE_LOW:
+            properties.append({'name': 'lowerSetpoint'})
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE_HIGH:
+            properties.append({'name': 'upperSetpoint'})
+        if supported & climate.SUPPORT_OPERATION_MODE:
+            properties.append({'name': 'thermostatMode'})
+        return properties
+
+    def properties_retrievable(self):
+        return True
+
+    def get_property(self, name):
+        if name == 'thermostatMode':
+            mode = self.entity.state.upper().replace('IDLE', 'OFF')
+            if mode not in CLIMATE_SUPPORTED_MODES:
+                return {
+                    'value': 'CUSTOM',
+                    'customName': mode,
+                }
+            return mode
+
+        temp = 0
+        unit = self.entity.attributes[CONF_UNIT_OF_MEASUREMENT]
+        if name == 'targetSetpoint':
+            temp = self.entity.attributes.get(ATTR_TEMPERATURE, 0)
+        elif name == 'lowerSetpoint':
+            temp = self.entity.attributes.get(climate.ATTR_TARGET_TEMP_LOW, 0)
+        elif name == 'upperSetpoint':
+            temp = self.entity.attributes.get(climate.ATTR_TARGET_TEMP_HIGH, 0)
+        else:
+            raise _UnsupportedProperty(name)
+
+        return {
+            'value': float(temp),
+            'scale': API_TEMP_UNITS[unit],
+        }
+
+
+def temperature_from_object(temp_obj, to_unit, absolute=True):
+    """Get temperature from Temperature object in requested unit."""
+    from_unit = TEMP_CELSIUS
+    temp = float(temp_obj['value'])
+
+    if temp_obj['scale'] == 'FAHRENHEIT':
+        from_unit = TEMP_FAHRENHEIT
+    elif ['scale'] == 'KELVIN':
```

Comment:
can this happen?

Reply:
https://developer.amazon.com/docs/device-apis/alexa-thermostatcontroller.html allows all Temperature objects in requests to to use `'scale': 'KELVIN'`, so we should be prepared to handle these requests.
(Typo here line 444 should be `elif temp_obj['scale'] == 'KELVIN':`

Reply:
Jeez Amazon, that's silly.